modules/git/hash.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
```

Comment:
Maybe we can rename the file name to `object_format.go` and `object_id.go`.

Reply:
Done